src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/entity/AssessmentStatus.kt

```diff
@@ -285,16 +138,436 @@ enum class AssessmentStatus {
 
   ;
 
-  fun allowsTransitionTo(state: AssessmentStatus): Boolean = transitions().map { it.to }.contains(state)
+  open fun tasks(): Map<UserRole, List<TaskProgress>> = emptyMap()
+}
 
-  open fun transitions(): Set<Transition> = emptySet()
+@Embeddable
+sealed interface AssessmentState {
+  val label: AssessmentStatus
 
-  open fun tasks(): Map<UserRole, List<TaskProgress>> = emptyMap()
+  object NotStarted : AssessmentState {
+    override val label = AssessmentStatus.NOT_STARTED
+  }
+
+  object EligibilityAndSuitabilityInProgress : AssessmentState {
+    override val label = AssessmentStatus.ELIGIBILITY_AND_SUITABILITY_IN_PROGRESS
+  }
+
+  object EligibleAndSuitable : AssessmentState {
+    override val label = AssessmentStatus.ELIGIBLE_AND_SUITABLE
+  }
+
+  object AwaitingAddressAndRiskChecks : AssessmentState {
+    override val label = AssessmentStatus.AWAITING_ADDRESS_AND_RISK_CHECKS
+  }
+
+  object AddressAndRiskChecksInProgress : AssessmentState {
+    override val label = AssessmentStatus.ADDRESS_AND_RISK_CHECKS_IN_PROGRESS
+  }
+
+  object AwaitingPreDecisionChecks : AssessmentState {
+    override val label = AssessmentStatus.AWAITING_PRE_DECISION_CHECKS
+  }
+
+  object AwaitingDecision : AssessmentState {
+    override val label = AssessmentStatus.AWAITING_DECISION
+  }
+
+  object Approved : AssessmentState {
+    override val label = AssessmentStatus.APPROVED
+  }
+
+  object AwaitingPreReleaseChecks : AssessmentState {
+    override val label = AssessmentStatus.AWAITING_PRE_RELEASE_CHECKS
+  }
+
+  object PassedPreReleaseChecks : AssessmentState {
+    override val label = AssessmentStatus.PASSED_PRE_RELEASE_CHECKS
+  }
+
+  object AddressUnsuitable : AssessmentState {
+    override val label = AssessmentStatus.ADDRESS_UNSUITABLE
+  }
+
+  object AwaitingRefusal : AssessmentState {
+    override val label = AssessmentStatus.AWAITING_REFUSAL
+  }
+
+  object IneligibleOrUnsuitable : AssessmentState {
+    override val label = AssessmentStatus.INELIGIBLE_OR_UNSUITABLE
+  }
+
+  object Refused : AssessmentState {
+    override val label = AssessmentStatus.REFUSED
+  }
+
+  object TimedOut : AssessmentState {
+    override val label = AssessmentStatus.TIMED_OUT
+  }
+
+  object Postponed : AssessmentState {
+    override val label = AssessmentStatus.POSTPONED
+  }
+
+  data class OptedOut(val previousState: AssessmentState) : AssessmentState {
+    override val label = AssessmentStatus.OPTED_OUT
```

Comment:
Stages can have all sorts of state, but it needs to be persisted so the stage can be reconstituted